### ISSUE TITLE
[apex] UnusedLocalVariable - false positive on case insensitivity allowed in Apex

### ISSUE THREAD

**Issue by mvenneman** [pmd/pmd#2626]
**Affects PMD Version:** 
6.24.0

**Rule:**
[UnusedLocalVariable](https://pmd.github.io/pmd-6.23.0/pmd_rules_apex_bestpractices.html#unusedlocalvariable)

**Description:**
Apex local variables are case-INsensitive.  The `UnusedLocalVariable` rule flags a variable declaration if that variable NOT used with the same case, but IS used with a different case.  Apex code will work as intended, and in Apex this is not an unused variable.

<https://github.com/pmd/pmd/blob/master/pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/bestpractices/UnusedLocalVariableRule.java>

**Code Sample demonstrating the issue:**

```
private String foo() {
   String Bar;
   bar = "bar";
   return bar; 
}
```

**Expected outcome:**
PMD should not report an UnusedLocalVariable violation in the above code sample, because `bar` is an acceptable reference to the originally declared `Bar` in Apex syntax.

In the rule's code, this can be addressed with another `if` statement in the `for` loop iterating over `potentialUsages`, such as:
```
if (usage.getImage() != null && usage.getImage().compareToIgnoreCase(variableName) == 0) {
   return data;
}
```
This issue cannot be addressed in the `ApexNode.hasImageEqualTo()` method because that method is used across languages where case-sensitivity is enforced (i.e. Java).

**Running PMD through:** 
CLI and "Apex PMD" VS Code extension

**Comment by mvenneman** [comment]
Label needs to be changed from bug to false-positive.

**Comment by oowekyala** [comment]
I wonder, whether it would make sense to make a rule to enforce that identifiers are always used in the same case. I'm not an apex developer, so I can't really tell
